### PR TITLE
feat(integrations): Vsts status sync outbound

### PR DESCRIPTION
**Pull request by lauryndbrown**

### CODE REVIEW

**Comment by lauryndbrown** [comment]
Checkout the Ecosystem Deliverables Status Sync Sentry -> Vsts for a demo video. Can't upload videos for whatever reason in github.

File: src/sentry/integrations/vsts/issues.py [hunk 1]
```diff
@@ -131,3 +131,27 @@ def sync_assignee_outbound(self, external_issue, user, assign=True, **kwargs):
...
                 }
             )
+
+    def sync_status_outbound(self, external_issue, is_resolved, project_id, **kwargs):
+        project_integration = ProjectIntegration.objects.get(
+            integration_id=external_issue.integration_id,
+            project_id=project_id,
+        )
+
+        status_name = 'resolve_status' if is_resolved else 'regression_status'
```
Comment (mnoble) [inline src/sentry/integrations/vsts/issues.py]: What does `regression_status` mean here?

File: src/sentry/integrations/vsts/issues.py [hunk 2]
```diff
@@ -131,3 +131,27 @@ def sync_assignee_outbound(self, external_issue, user, assign=True, **kwargs):
...
+                self.instance, external_issue.key, state=status)
+        except (ApiUnauthorized, ApiError):
+            self.logger.info(
+                'vsts.failed-to-change-status',
+                extra={
+                    'integration_id': external_issue.integration_id,
+                    'is_resolved': is_resolved,
+                    'issue_key': external_issue.key,
+                }
+            )
```
Comment (mnoble) [inline src/sentry/integrations/vsts/issues.py]: We should probably add a simple test for this.

File: src/sentry/integrations/example/integration.py [hunk 3]
```diff
@@ -65,7 +65,8 @@ def get_issue(self, issue_id):
     def sync_assignee_outbound(self, external_issue, user, assign=True, **kwargs):
         pass
 
-    def sync_status_outbound(external_issue, is_resolved):
+    def sync_status_outbound(external_issue, is_resolved, project_id):
+        # TODO(lb): should this have self?
```
Comment (mnoble) [inline src/sentry/integrations/example/integration.py]: Based on the actual class, I think so, yea.

Comment (lauryndbrown) [reply src/sentry/integrations/example/integration.py]: Cool. I think jess just might have been in a hurry. :)

File: src/sentry/integrations/vsts/issues.py (hunk 1 above)
Comment (mnoble) [reply src/sentry/integrations/vsts/issues.py]: Oh I see, "resolved" and "regression" are the only two statuses we have, so they're the only two options to sync. Got it.

File: src/sentry/integrations/vsts/issues.py [hunk 4]
```diff
@@ -131,3 +131,27 @@ def sync_assignee_outbound(self, external_issue, user, assign=True, **kwargs):
...
+        project_integration = ProjectIntegration.objects.get(
+            integration_id=external_issue.integration_id,
+            project_id=project_id,
+        )
+
+        status_name = 'resolve_status' if is_resolved else 'regression_status'
+        try:
+            status = project_integration.config[status_name]
+        except KeyError:
+            return  # TODO(lb): what's the expected behavior here?
```
Comment (EvanPurkhiser) [inline src/sentry/integrations/vsts/issues.py]: I imagine do nothing, if it's not configured. Maybe we can later add some kind of error reporting later, but this seems fine

File: src/sentry/integrations/vsts/issues.py [hunk 5]
```diff
@@ -131,3 +131,27 @@ def sync_assignee_outbound(self, external_issue, user, assign=True, **kwargs):
...
+            self.get_client().update_work_item(
+                self.instance, external_issue.key, state=status)
+        except (ApiUnauthorized, ApiError):
+            self.logger.info(
+                'vsts.failed-to-change-status',
+                extra={
+                    'integration_id': external_issue.integration_id,
+                    'is_resolved': is_resolved,
+                    'issue_key': external_issue.key,
+                }
```
Comment (EvanPurkhiser) [inline src/sentry/integrations/vsts/issues.py]: Might want to add the api error if we can.

**Review by evanpurkhiser — APPROVED** [review]
looks pretty good to me

**Pull request merged**